internal/exec/stages/fetch_offline/fetch_offline_test.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2020 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fetch_offline
+
+import (
+	"testing"
+
+	"github.com/coreos/ignition/v2/config/util"
+	"github.com/coreos/ignition/v2/config/v3_2_experimental/types"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func checkNeedsNet(t *testing.T, cfg *types.Config) bool {
+	needsNet, err := configNeedsNet(cfg)
+	assert.Equal(t, err, nil, "unexpected error: %v", err)
+	return needsNet
+}
+
+func assertNeedsNet(t *testing.T, cfg *types.Config) {
+	assert.Equal(t, checkNeedsNet(t, cfg), true, "unexpected config dosn't needs net: %v", *cfg)
+}
+
+func assertNotNeedsNet(t *testing.T, cfg *types.Config) {
+	assert.Equal(t, checkNeedsNet(t, cfg), false, "unexpected config needs net: %v", *cfg)
+}
+
+func TestConfigNeedsNet(t *testing.T) {
+	cfg := types.Config{
+		Ignition: types.Ignition{
+			Version: "3.2.0-experimental",
+			Config: types.IgnitionConfig{
+				Replace: types.Resource{
+					Source: util.StrToPtr("http://example.com/config.ign"),
+				},
+			},
+		},
+	}
+	assertNeedsNet(t, &cfg)
+	cfg.Ignition.Config.Replace.Source = nil
```

Comment:
Could we test a merged/replaced config from a data URL which contains a network dependency?

Reply:
I looked into this, but the issue is that config rendering is part of the engine code, so the introspection code won't pick this up. I think it'd require moving this to a blackbox test and tweaking the blackbox test harness so that we can run the "fetch-offline" (and "fetch" too for that matter) stages explicitly. This would go well with https://github.com/coreos/ignition/issues/950 too; mind if we punt on this for now?

(That said, I *did* verify that this case is covered.)